buffer.go

```diff
@@ -384,7 +385,13 @@ func (b *Buffer) Read(data []byte, c Cursor) (n int, next Cursor, err error) {
 		return b.readFirst(data)
 	}
 
-	return b.readOffset(data, offset)
+	n1, c1, err := b.readOffset(data, offset)
```

Comment:
There's not a need to panic here, I think. I would leave the code as it was before

Reply:
maybe set err instead of panic and return it

Reply:
@Dorokhov I wonder where the fix is. If it is the panic, then how do you test it?

Reply:
done